source/plugin/blocks/accessing.rst

```diff
@@ -34,56 +34,104 @@ referenced block is any kind of banner by checking the blocks type.
-redstone power). The following example checks if a block is a regular chest directly or indirectly powered with redstone.
+The data of a block is held as a ``DataManipulator``, as similar with other parts of the API. This is the container that holds information on components of our block such as the orientation of a block, specific types (stone vs. granite), and so on. Checking the values of these manipulators is easy, you just need to check the block's direction ``DirectionalData``.
 
 .. code-block:: java
 
-    public boolean isPoweredChest(Location block) {
-        if (block.getBlock().getType() == BlockTypes.CHEST) {
-            return block.isBlockPowered() || block.isBlockIndirectlyPowered();
-        } else {
+    import org.spongepowered.api.data.key.Keys;
+    import org.spongepowered.api.data.manipulator.mutable.block.DirectionalData;
+
+    public boolean isFacingNorth(Location<World> blockLoc) {
+        Optional<DirectionalData> optionalData = blockLoc.get(DirectionalData.class);
+        if (!optionalData.isPresent()) {
             return false;
         }
+        DirectionalData data = optionalData.get();
+        if (data.get(Keys.DIRECTION).get().equals(Direction.NORTH)) {
+            return true;
+        }
+        return false;
     }
 
+First, we need to know which ``DataManipulator`` sub-interface we need. Those that are applicable to blocks are found in
+the ``org.spongepowered.api.data.manipulator.mutable`` and ``org.spongepowered.api.data.manipulator.mutable.block`` packages. Then we
+can just pass that class to the ``get()`` method on ``Location`` which will return an ``Optional``. We then have to check if our ``DataManipulator`` actually exists for our block by checking ``ifPresent()``. If it exists, then we can use it. 
+
+More on ``DataManipulator`` s can be found in the :doc:`data documentation <../data/datamanipulators>`.
+
+.. tip ::
+    
+    If a block is guaranteed to have a certain ``DataManipulator`` such as ``DirectionalData`` with stairs, then there is no need to check for ``isPresent()``. Just remove the optional around the ``DataManipulator`` and fetch the non-optional data by adding ``.get()`` to the end of the statement.
+
 Block States
 ~~~~~~~~~~~~
 
-Additional Data (like orientation, subtypes like granite vs regular stone, farm land moisture ... ) can be obtained from
-a ``BlockState``. A ``BlockState`` contains a ``BlockType`` and further data that can be accessed via a set of data
-manipulators that all extend the ``DataManipulator`` interface. The following example will return ``true`` if a the block
-at a given ``Location`` is a) a Sponge and b) wet, false otherwise.
+A ``BlockState`` contains a ``BlockType``,  any ``DataManipulator`` s and properties that are applied to the block, and any ``BlockTrait`` s for a block. It stores all immutable value's for a particular block. An use is of this is getting an  ``ImmutableDataManipulator``, as shown below:
```

Comment:
Yeah I threw this whole thing together just today and after a while some things got mixed up as I changed wording to make things fit.